### PR TITLE
Internal: Update Nested Tabs HTML for Menu A11Y [ED-11681]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js [hunk 1]
```diff
@@ -98,18 +98,20 @@ export default class NestedTabsHtml extends Base {
 		this.setSettings( originalToggleMethods );
 	}
 
-	deactivateActiveTab( tabIndex ) {
+	deactivateActiveTab( newTabIndex ) {
 		const settings = this.getSettings(),
 			activeClass = settings.classes.active,
-			activeTitleFilter = !! tabIndex ? this.getTabTitleFilterSelector( tabIndex ) : settings.ariaAttributes.activeTitleSelector,
```
Comment (hein-obox) [inline modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js:104]: The 'tabIndex' parameter was never used in our code inside the 'deactivateActiveTab()' method.

File: modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js [hunk 2]
```diff
@@ -321,8 +325,12 @@ export default class NestedTabsHtml extends Base {
 		}
 	}
 
-	changeActiveTabByKeyboard( event, tabIndex ) {
-		this.changeActiveTab( tabIndex, true );
+	changeActiveTabByKeyboard( event, settings ) {
+		if ( settings.widgetId !== this.getID() ) {
```
Comment (hein-obox) [inline modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js:329]: Verify that this is triggered by the correct widget.

File: assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js [hunk 3]
```diff
@@ -55,43 +72,79 @@ export default class NestedTitleKeyboardHandler extends Base {
...
+			const isLinkElement = 'a' === event.currentTarget.tagName.toLowerCase();
+
+			if ( ! elementorFrontend.isEditMode() && isLinkElement ) {
+				location.href = event.currentTarget.getAttribute( 'href' );
+				event.stopPropagation();
+			}
+
+			if ( isLinkElement ) {
+				return;
+			}
```
Comment (davseve) [inline assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js:138]: Suggestion
Nested conditions.
Can this be extracted to a separate function?

File: assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
-const directionNext = 'next',
-	directionPrevious = 'previous';
-
 export default class NestedTitleKeyboardHandler extends Base {
+	__construct( settings ) {
+		super.__construct( settings );
+
+		this.directionNext = 'next';
+		this.directionPrevious = 'previous';
+		this.focusableElementSelector = 'audio, button, canvas, details, iframe, input, select, summary, textarea, video, [accesskey], [contenteditable], [href], [tabindex]:not([tabindex="-1"])';
```
Comment (davseve) [inline assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js:9]: Nice!
Loved this dictionary!!

**Review by davseve — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js (hunk 3 above)
Comment (davseve) [inline assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js:138]: Suggestion
Nested conditions.
Can this be extracted to a separate function?

**Review by davseve — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js (hunk 3 above)
Comment (hein-obox) [reply assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js:138]: Updated

**Review by davseve — DISMISSED** [review]

File: assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js [hunk 5]
```diff
@@ -55,41 +72,81 @@ export default class NestedTitleKeyboardHandler extends Base {
...
+			this.handleTitleEscapeKeyEvents( event );
 		}
 	}
 
-	getTitleEvents() {
-		return {
-			keydown: this.handleTitleKeyboardNavigation.bind( this ),
-		};
+	handeTitleLinkEnterOrSpaceEvent( event ) {
+		const isLinkElement = 'a' === event.currentTarget.tagName.toLowerCase();
```
[the remark this answers is not in the document]
Comment (louiswol94) [reply assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js:142]: @hein-obox Verify this everywhere, that you might not possible try to access a property that does not exist. Just to be safe - this is where both these issues come from 😄 

https://github.com/orgs/elementor/discussions/22894
https://wordpress.org/support/topic/exception-thrown-when-changing-device-in-devtools/

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js (hunk 2 above)
Comment (hein-obox) [reply modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js:329]: I am using this 'trigger' event in here assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js: 

elementorFrontend.elements.$window.trigger( 'elementor/nested-elements/activate-by-keyboard', { widgetId: this.getID(), titleIndex } );

This triggers every function that is listening to this trigger and can trigger an incorrect Tabs or Menu widget.

**Comment by hein-obox** [comment]
https://github.com/elementor/elementor/pull/23389#discussion_r1293317797

I am using this 'trigger' event in here assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js:

elementorFrontend.elements.$window.trigger( 'elementor/nested-elements/activate-by-keyboard', { widgetId: this.getID(), titleIndex } );

This triggers every function that is listening to this trigger and can trigger an incorrect Tabs or Menu widget.

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**